### ISSUE TITLE
Subsequent env variables are ignored

### ISSUE THREAD

**Issue by olaf-cichocki** [mucsi96/nightwatch-api#224]
## Expected Behavior

Stoping session and webdriver, and starting them both again with new env variable should make it use new env settings.

## Current Behavior

When stopping both the session and webdriver, and starting them with new env, settings doesn't change. It 'locks' on the first one used, and then on each subsequent start, it uses settings from the first run.

## Steps to Reproduce (for bugs)

We use the code below, to start and stop the webdriver, we have separate settings for multiple different devices inside our config.
```
Given(/I open browser on (.*)/, async (deviceName) => {
  await startWebDriver({ env: deviceName }).catch(err => console.log(err));
  await createSession({ env: deviceName });
});

After(async () => {
  await closeSession();
  await stopWebDriver();
})
```
We set separate chromeOptions for each of them:
```
chromeOptions: {
  args: [
    `window-size=${device.viewport.width},${device.viewport.height}`,
    `user-agent=${device.userAgent}`,
    'headless',
    'disable-gpu',
  ],
  mobileEmulation: {
    deviceMetrics: {
      userAgent: device.userAgent,
      width: device.viewport.width,
      height: device.viewport.height
    }
  },
},
```
If we run test, the env from first device tested is then used for all tests (when checking desiredCapabilities from inside of the test, they all look the same, instead of using different one for each test). I checked and we are passing correct env on each test.

## Context

We have integration with an external solution that uses user agent and window size for feature detection, and both HTML and behaviour differs depending on the device. We want to test all cases.

## Your Environment

- Version used: nightwatch: `^1.0.18`, nightwatch-api: `latest`, cucumber: `^5.1.0`
- Browser Name and version: Chromedriver `2.46.0`
- Operating System and version (desktop or mobile): Windows 10 Home

**Comment by mucsi96** [comment]
Hi @olaf-cichocki! Thx for you feedback and issue with all needed details. I will try to look into it.

**Comment by olaf-cichocki** [comment]
Thank you @mucsi96! If there is anything else I can provide to make debugging this easier, please let me know.

**Comment by mucsi96** [comment]
Hi, Thanks for you patient. I have identified the root case. The issue is that we are caching the Nightwatch configuration instance and never invalidate the cache. A possible solution would be cache based on environment name. If that changes we create a new Nightwatch confirmation instance. I plan to reproduce the issue with some unit tests to prevet similar issue in the future and release a fix this week if possible.

**Comment by mucsi96** [comment]
Fix released in 2.1.2
